File: src/main/java/com/birdy/blogbackend/domain/enums/UserStatus.java
package com.birdy.blogbackend.domain.enums;

import lombok.Getter;

/**
 * @author birdy
 */

@Getter
public enum UserStatus {
    /**
     * 正常
     */
    NORMAL(0, "正常"),
    /**
     * 禁用
     */
    BANNED(1, "禁用"),
    /**
     * 删除
     */
    DELETED(2, "删除");

    final int code;
    final String desc;

    UserStatus(int code, String desc) {
        this.code = code;
        this.desc = desc;
    }

    public static UserStatus valueOf(int value) {
        for (UserStatus status : UserStatus.values()) {
            if (status.code == value) {
                return status;
            }
        }
        return BANNED;
    }
}

File: src/main/kotlin/com/birdy/blogbackend/domain/enums/UserStatus.kt
package com.birdy.blogbackend.domain.enums

import lombok.Getter

/**
 * @author birdy
 */
@Getter
enum class UserStatus(val code: Int, val desc: String) {
    /**
     * 正常
     */
    NORMAL(0, "正常"),

    /**
     * 禁用
     */
    BANNED(1, "禁用"),

    /**
     * 删除
     */
    DELETED(2, "删除");

    companion object {
        fun valueOf(value: Int): UserStatus {
            for (status in entries) {
                if (status.code == value) {
                    return status
                }
            }
            return BANNED
        }
    }
}
